api/bertyprotocol.proto

```diff
@@ -42,6 +42,9 @@ service ProtocolService {
   // ContactUnblock unblocks a contact from sending requests
   rpc ContactUnblock (types.v1.ContactUnblock.Request) returns (types.v1.ContactUnblock.Reply);
 
+  // WaitMemberIsReady returns a stream that will send a event as soon as the member is ready to receive messages.
+  rpc WaitMemberIsReady (types.v1.WaitMemberIsReady.Request) returns (stream types.v1.WaitMemberIsReady.Reply);
```

Comment:
This is a bad idea for several reasons:
- It's counter-intuitive and complicates the task for the protocol user, if something like this were to be implemented (waiting for secrets to be sent before sending messages), it would have to be internal to the protocol (using a cache or something), especially since it would have to be systematic.
- It's useless in the sense that the member must be able to receive secrets and messages out of order and have the ability to replay the messages once the secret is received.
- It's also useless as long as the member doesn't send an ACK, the only thing that seems to be done in this call is to check if the "sending" was done in orbit-db. The member has no guarantee to receive the messages before the secret despite this function.

Reply:
I'll answer your three points individually:

- yes, the only solution is to set a cache for messages before sending them if secrets are not sent.
- I don't agree because replay will not work, because messages sent before secrets sents can never be opened because the sender will send the chainkey already derived.
- That's why there is a message cache for the receiver.

Reply:
Conclusion after an audio call with D4ryll00:
- This API method in fact does not wait until the member is ready / has received the secrets, it ensures that messages are not sent until the secrets have been sent to prevent the chain key from being derived by sending messages before the chain key is sent as a secret (messages sent before will therefore be undecryptable).
- This API method will be removed and replaced by an internal lock.